libfreerdp/core/connection.c

```diff
@@ -184,8 +185,19 @@ static BOOL rdp_client_reset_codecs(rdpContext* context)
 	if (!context->codecs)
 		return FALSE;
 
-	return freerdp_client_codecs_prepare(context->codecs, FREERDP_CODEC_ALL,
-	                                     settings->DesktopWidth, settings->DesktopHeight);
+	if (settings->RemoteFxCodec == FALSE)
```

Comment:
What is this for?

Reply:
to avoid initialization of codecs->rfx, codecs->nsc, if it disabled in settings, at best it does memory allocations, at worst, creating threads, mutexes, i think it does not make sense if it is unused at all and also harmful for resources

Reply:
Ok, then it would be best to create a function `DWORD freerdp_settings_codec_mask(const rdpSettings* settings)` or similar to get the mask and use it instead of duplicating the whole code.